src/event/sys/windows/parse.rs

```diff
@@ -61,19 +61,16 @@ pub(crate) fn handle_key_event(
 }
 
 fn handle_surrogate(surrogate_buffer: &mut Option<u16>, new_surrogate: u16) -> Option<char> {
```

Comment:
This method could do with an explanation that clarifies the intent as well as some unit tests for the same.

Reply:
Added test.